src/vm/i386/jitinterfacex86.cpp

```diff
@@ -1349,17 +1241,6 @@ void InitJITHelpers1()
         SetJitHelperFunction(CORINFO_HELP_NEWARR_1_VC, pMethodAddresses[4]);
         pMethodAddresses[5] = JIT_TrialAlloc::GenAllocArray((JIT_TrialAlloc::Flags)(flags|JIT_TrialAlloc::ALIGN8));
         SetJitHelperFunction(CORINFO_HELP_NEWARR_1_ALIGN8, pMethodAddresses[5]);
-
-        fastObjectArrayAllocator = (FastObjectArrayAllocatorFuncPtr)JIT_TrialAlloc::GenAllocArray((JIT_TrialAlloc::Flags)(flags|JIT_TrialAlloc::NO_FRAME|JIT_TrialAlloc::OBJ_ARRAY));
-        fastPrimitiveArrayAllocator = (FastPrimitiveArrayAllocatorFuncPtr)JIT_TrialAlloc::GenAllocArray((JIT_TrialAlloc::Flags)(flags|JIT_TrialAlloc::NO_FRAME));
-
-        // If allocation logging is on, then we divert calls to FastAllocateString to an Ecall method, not this
-        // generated method. Find this workaround in Ecall::Init() in ecall.cpp.
-        ECall::DynamicallyAssignFCallImpl((PCODE) JIT_TrialAlloc::GenAllocString(flags), ECall::FastAllocateString);
```

Comment:
I think you want to keep this one. It is just the NO_FRAME ones that can go.

Reply:
Ok. It was tempting since it was becoming the last user of GenAllocString.

---
In reply to: [280182020](https://github.com/dotnet/coreclr/pull/24096#discussion_r280182020) [](ancestors = 280182020)

Reply:
GenAllocString is back, but `NO_FRAME` is out since noone uses it now.

---
In reply to: [280182659](https://github.com/dotnet/coreclr/pull/24096#discussion_r280182659) [](ancestors = 280182659,280182020)

Reply:
So add back this line?

Reply:
OOps. I thought, I've added it. 

---
In reply to: [280527577](https://github.com/dotnet/coreclr/pull/24096#discussion_r280527577) [](ancestors = 280527577)

Reply:
Also needs to rebase to pick conflicting changes in GC.cs

---
In reply to: [280530904](https://github.com/dotnet/coreclr/pull/24096#discussion_r280530904) [](ancestors = 280530904,280527577)